src/test/java/com/amazon/ion/impl/EncodingDirectiveCompilationTest.java

```diff
@@ -867,6 +897,88 @@ public void macroInvocationsNestedWithinParameter(InputType inputType, StreamTyp
         }
     }
 
+    /**
+     * Counts the number of times the given substring occurs in the given string.
+     * @param string the string.
+     * @param substring the substring.
+     * @return the number of occurrences.
+     */
+    private static int countOccurrencesOfSubstring(String string, String substring) {
+        int lastMatchIndex = 0;
+        int count = 0;
+        while (lastMatchIndex >= 0) {
+            lastMatchIndex = string.indexOf(substring, lastMatchIndex);
+            if (lastMatchIndex >= 0) {
+                lastMatchIndex += substring.length();
+                count++;
+            }
+        }
+        return count;
+    }
+
+    /**
+     * Performs a macro-aware transcode of the given data, verifying that the resulting stream has the
+     * characteristics described by the arguments to this method and that it is data-model equivalent
+     * to the source data.
+     * @param data the source data.
+     * @param inputType the InputType to test.
+     * @param streamType the StreamType to which the source data will be transcoded.
+     * @param expectedNumberOfIonVersionMarkers the expected number of Ion version markers in the transcoded stream (text only).
+     * @param expectedNumberOfAddSymbolsInvocations the expected number of add_symbols invocations in the transcoded stream (text only).
+     * @param expectedNumberOfAddMacrosInvocations the expected number of add_macros invocations in the transcoded stream (text only).
+     * @param expectedNumberOfSetSymbolsInvocations the expected number of set_symbols invocations in the transcoded stream (text only).
+     * @param expectedNumberOfSetMacrosInvocations the expected number of set_macros invocations in the transcoded stream (text only).
+     * @param expectedNumberOfExplicitEncodingDirectives the expected number of $ion_encoding occurrences in the transcoded stream (text only).
+     */
+    private void verifyMacroAwareTranscode(
+        byte[] data,
+        InputType inputType,
+        StreamType streamType,
+        int expectedNumberOfIonVersionMarkers,
+        int expectedNumberOfAddSymbolsInvocations,
+        int expectedNumberOfAddMacrosInvocations,
+        int expectedNumberOfSetSymbolsInvocations,
+        int expectedNumberOfSetMacrosInvocations,
+        int expectedNumberOfExplicitEncodingDirectives
```

Comment:
This is ugly; open to suggestions.

Reply:
Here's a suggestion using a custom matcher: https://gist.github.com/jobarr-amzn/b009638a868941d63c4f1cf8277a7dfc#file-encodingdirectivecompilationtest-java-L925-L1013

Reply:
Thanks, done